framework_tvb/tvb/interfaces/rest/server/resources/operation/operation_resource.py

```diff
@@ -117,10 +121,24 @@ def post(self, project_gid, algorithm_module, algorithm_classname):
                                                                  {})
             storage_path = self.files_helper.get_project_folder(project, str(operation.id))
 
+            if isinstance(adapter_instance, ABCUploader):
+
+                index = 1
+                for key, value in adapter_instance.get_upload_information().items():
+                    data_file = self.extract_file_from_request(file_name='data_file_' + str(index), file_extension=value)
```

Comment:
Can;t you already use the 'key' from the Uploader. get_upload_information on the client SDK also?